gitpod-io/gitpod
# [server] allow fetching task specific prebuild-logs

## Description

Expose Prebuild Logs per task.

<img width="1399" alt="Screenshot 2024-05-30 at 16 22 05" src="https://github.com/gitpod-io/gitpod/assets/372735/0d3b42d9-acd5-4121-8b6e-c374131922a4">

## Related Issue(s)

See https://linear.app/gitpod/issue/ENT-172/prebuilds-show-logs-of-all-tasks

## How to test


Join my org https://se-prebuild-by-task.preview.gitpod-dev.com/orgs/join?inviteId=07ebd814-a02f-4aeb-abf3-c87f351bec47

Check [the prebuild details](https://se-prebuild-by-task.preview.gitpod-dev.com/prebuilds/ab554c76-8fc8-41fa-99ed-6017908aa9b5) and open the network tab in chrome devtools.

## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
        <li><b>🏷️ Name</b> - se-prebuild-by-task</li>
        <li><b>🔗 URL</b> - <a href="https://se-prebuild-by-task.preview.gitpod-dev.com/workspaces" target="_blank">se-prebuild-by-task.preview.gitpod-dev.com/workspaces</a>.</li>
        <li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
        <li><b>📦 Version</b> - </li>
        <li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-se-prebuild-by-task%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
